### PR TITLE
[LYN-7520] Wildcard Source Dependencies include files in cache/excluded files

### PR DESCRIPTION
**Pull request by amzn-mike**
Added a cache that builds a list of all watched folders that are ignored.
Added a FindWildcardMatches version which takes a list of ignored directories and will not scan any deeper if it finds one.
Updated wildcard dependency resolution code to take ignored folders into account

### CODE REVIEW

File: Code/Tools/AssetProcessor/native/AssetManager/FileStateCache.cpp [hunk 1]
```diff
@@ -7,9 +7,10 @@
  */
 
 #include "FileStateCache.h"
-#include "native/utilities/assetUtils.h"
-#include <AssetProcessor_Traits_Platform.h>
 
+#include <AssetProcessor_Traits_Windows.h>
```
Comment (jackalbe) [inline Code/Tools/AssetProcessor/native/AssetManager/FileStateCache.cpp:11]: it seems odd to see a platform specific include like AssetProcessor_Traits_Windows.h, right?

Comment (amzn-mike) [reply Code/Tools/AssetProcessor/native/AssetManager/FileStateCache.cpp:11]: I thought that too.  So I removed it, then the compile failed and pointed out why.  There's a define check to see if we're on a case sensitive file system.  On case-insensitive filesystems the path key is lowercased before storing it

Comment (jackalbe) [reply Code/Tools/AssetProcessor/native/AssetManager/FileStateCache.cpp:11]: I thought there was a AssetProcessor_Traits.h that included the proper PAL header file. No?

Comment (jackalbe) [reply Code/Tools/AssetProcessor/native/AssetManager/FileStateCache.cpp:11]: AssetProcessor_Traits_Platform.h

**Review by AMZN-stankowi — APPROVED** [review]

File: Code/Tools/AssetProcessor/native/AssetManager/FileStateCache.cpp (hunk 1 above)
Comment (amzn-mike) [reply Code/Tools/AssetProcessor/native/AssetManager/FileStateCache.cpp]: Ah, ended up including the wrong thing when I put it back.  Fixed!

**Review by jackalbe — APPROVED** [review]

**Pull request merged**